src/main/scala/edu/ie3/util/geo/GeoUtils.scala

```diff
@@ -0,0 +1,341 @@
+  }
+
+  /** Calculates and orders the coordinate distances from a base coordinate to a
+    * list of coordinates
+    *
+    * @param baseCoordinate
+    *   the base coordinate
+    * @param coordinates
+    *   the coordinates to calculate the distance for
+    * @return
+    *   a sorted set of [[CoordinateDistance]]
+    */
+  def calcOrderedCoordinateDistances(
+      baseCoordinate: Point,
+      coordinates: Array[Point]
+  ): SortedSet[CoordinateDistance] = {
+    val coordinateDistances = coordinates.map(coordinate =>
+      new CoordinateDistance(baseCoordinate, coordinate)
+    )
+    TreeSet(coordinateDistances: _*)
+  }
+
+  /** Calculates the great circle distance between two coordinates
+    *
+    * @param latA
+    *   latitude of coordinate a
+    * @param longA
+    *   longitude of coordinate a
+    * @param latB
+    *   latitude of coordinate b
+    * @param longB
+    *   longitude of coordinate b
+    * @return
+    *   the distance between the coordinates as a quantity
+    */
+  def calcHaversine(
+      latA: Double,
+      longA: Double,
+      latB: Double,
+      longB: Double
+  ): ComparableQuantity[Length] = {
+    val r = EARTH_RADIUS
+    val dLat = Quantities.getQuantity(toRadians(latB - latA), RADIAN)
+    val dLon = Quantities.getQuantity(toRadians(longB - longA), RADIAN)
+    val a = pow(sin(dLat.getValue.doubleValue / 2), 2) + cos(
+      toRadians(latA)
+    ) * cos(toRadians(latB)) * pow(sin(dLon.getValue.doubleValue / 2), 2)
+    val c = 2 * atan2(sqrt(a), sqrt(1 - a))
+    r.multiply(c)
+  }
+
+  /** Builds a convex hull from a set of latitude/longitude coordinates.
+    *
+    * @param coordinates
+    *   the coordinates to consider
+    * @return
+    *   a Try of the resulting polygon
+    */
+  def buildConvexHull(coordinates: Set[Coordinate]): Try[Polygon] = {
+    val projectedCoordinates = coordinates
+    new ConvexHull(
+      projectedCoordinates.toArray,
+      DEFAULT_GEOMETRY_FACTORY
+    ).getConvexHull match {
+      case projectedPolygon: Polygon =>
+        Success(buildPolygon(projectedPolygon.getCoordinates))
+      case _: LineString =>
+        Failure(
+          new GeoException(
+            s"Got a line string as a convex hull. Probably coordinates: $coordinates only contains two different coordinates."
+          )
+        )
+      case _: Point =>
+        Failure(
+          new GeoException(
+            s"Got a point as a convex hull. Probably coordinates: $coordinates only contains one coordinate."
+          )
+        )
+      case _: GeometryCollection =>
+        Failure(
+          new GeoException(
+            s"Got a GeometryCollection. Probably $coordinates was empty."
+          )
+        )
+      case _ => Failure(new GeoException(s"Got an unexpected return type."))
+    }
+  }
+
+  def buildPoint(lat: Double, long: Double): Point = {
+    buildPoint(buildCoordinate(lat, long))
+  }
+
+  def buildPoint(coordinate: Coordinate): Point = {
```

Comment:
This is doubled with your definition in the rich quantity. Please adapt the method and as well point out with a deprecation notice, that the rich class may be used.